src/gluonts/model/seasonal_agg/_predictor.py

```diff
@@ -0,0 +1,130 @@
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+from typing import Callable
+
+import numpy as np
+
+from gluonts.core.component import validated
+from gluonts.dataset.common import DataEntry
+from gluonts.dataset.util import forecast_start
+from gluonts.dataset.field_names import FieldName
+from gluonts.model.forecast import Forecast, SampleForecast
+from gluonts.model.predictor import RepresentablePredictor
+from gluonts.transform.feature import (
+    LastValueImputation,
+    MissingValueImputation,
+)
+
+
+class SeasonalAggregatePredictor(RepresentablePredictor):
+    """
+    Seasonal aggegate forecaster.
+
+    For each time series :math:`y`, this predictor produces a forecast
+    :math:`\\tilde{y}(T+k) = f\big(y(T+k-h), y(T+k-2h), ...,
+    y(T+k-mh)\big)`, where :math:`T` is the forecast time,
+    :math:`k = 0, ...,` `prediction_length - 1`, :math:`m =`num_seasons`,
+    :math:`h =`season_length` and :math:`f =`agg_fun`.
+
+    If `prediction_length > season_length` :math:\times `num_seasons`, then the
+    seasonal aggregate is repeated multiple times. If a time series is shorter
+    than season_length` :math:\times `num_seasons`, then the `agg_fun` is
+    applied to the full time series.
+
+    Parameters
+    ----------
+    prediction_length
+        Number of time points to predict.
+    season_length
+        Seasonality used to make predictions.
+    num_seasons
+        Number of seasons to aggregate.
+    agg_fun
+        Aggregate function.
+    imputation_method
+        The imputation method to use in case of missing values.
+        Defaults to :py:class:`LastValueImputation` which replaces each missing
+        value with the last value that was not missing.
+    """
+
+    @validated()
+    def __init__(
+        self,
+        prediction_length: int,
+        season_length: int,
+        num_seasons: int,
+        agg_fun: Callable = np.nanmean,
+        imputation_method: MissingValueImputation = LastValueImputation(),
+    ) -> None:
+        super().__init__(prediction_length=prediction_length)
+
+        assert (
+            not isinstance(season_length, int) or season_length > 0
+        ), "The value of `season_length` should be > 0"
+
+        assert (
+            not isinstance(num_seasons, int) or num_seasons > 0
+        ), "The value of `num_seasons` should be > 0"
+
+        self.prediction_length = prediction_length
+        self.season_length = season_length
+        self.num_seasons = num_seasons
+        self.agg_fun = agg_fun
+        self.imputation_method = imputation_method
+
+    def predict_item(self, item: DataEntry) -> Forecast:
+        if isinstance(self.season_length, int):
+            season_length = self.season_length
+        else:
+            season_length = self.season_length(item["start"].freq)
```

Comment:
Can `self.season_length` be a callable? Its type in the signature is `int`.

Reply:
yes, updated the signature.